juju/model.py

```diff
@@ -1055,7 +1055,7 @@ def predicate(delta):
 
         # Submit the request.
         client_facade = client.ClientFacade.from_connection(self.connection())
-        results = await client_facade.AddMachines([params])
+        results = await client_facade.AddMachines(params=[params])
```

Comment:
why are we list-ifying params the object?

Reply:
params here is the template definition for 1 machine. AddMachines is a bulk API that would let you supply multiple templates at the same time. The shim is doing "add_machine" to add a single machine.